docs/docs/sophia_stdlib.md

```diff
@@ -0,0 +1,2069 @@
+
+
+#### default
+```
+Option.default(def : 'a, o : option('a)) : 'a
+```
+
+Escapes `option` wrapping by providing default value for `None`.
+
+
+#### force
+```
+Option.force(o : option('a)) : 'a
+```
+
+Forcefully escapes `option` wrapping assuming it is `Some`. Throws error on `None`.
+
+
+#### on_elem
+```
+Option.on_elem(o : option('a), f : 'a => unit) : unit
+```
+
+Evaluates `f` on element under `Some`. Does nothing on `None`.
+
+
+#### map
+```
+Option.map(f : 'a => 'b, o : option('a)) : option('b)
+```
+
+Maps element under `Some`. Leaves `None` unchanged.
+
+
+#### map2
+```
+Option.map2(f : ('a, 'b) => 'c, o1 : option('a), o2 : option('b)) : option('c)
+```
+
+Applies arity 2 function over two `option`s' elements. Returns `Some` iff both of `o1` and `o2` were `Some`, or `None` otherwise. For instance
+```
+map2((a, b) => a + b, Some(1), Some(2))
+```
+will yield `Some(3)` and
+```
+map2((a, b) => a + b, Some(1), None)
+```
+will yield `None`.
+
+
+#### map3
+```
+Option.map3(f : ('a, 'b, 'c) => 'd, o1 : option('a), o2 : option('b), o3 : option('c)) : option('d)
+```
+
+Same as [map2](#map2) but with arity 3 function.
+
+
+#### app_over
+```
+Option.app_over(f : option ('a => 'b), o : option('a)) : option('b)
+```
+
+Applies function under `option` over argument under `option`. If either of them is `None` the result will be `None` as well. For instance
+```
+app_over(Some((x) => x + 1), Some(1))
+```
+will yield `Some(2)` and
+```
+app_over(Some((x) => x + 1), None)
+```
+will yield `None`.
+
+
+#### flat_map
+```
+Option.flat_map(f : 'a => option('b), o : option('a)) : option('b)
+```
+
+Performs monadic bind on an `option`. Extracts element from `o` (if present) and forms new `option` from it. For instance
+```
+flat_map((x) => Some(x + 1), Some(1))
+```
+will yield `Some(2)` and
+```
+flat_map((x) => Some(x + 1), None)
+```
+will yield `None`.
+
+
+#### to_list
+```
+Option.to_list(o : option('a)) : list('a)
+```
+
+Turns `o` into an empty (if `None`) or singleton (if `Some`) list.
+
+
+#### filter_options
+```
+Option.filter_options(l : list(option('a))) : list('a)
+```
+
+Removes `None`s from list and unpacks all remaining `Some`s. For instance
+```
+filter_options([Some(1), None, Some(2)])
+```
+will yield `[1, 2]`.
+
+
+#### seq_options
+```
+Option.seq_options(l : list (option('a))) : option (list('a))
+```
+
+Tries to unpack all elements of a list from `Some`s. Returns `None` if at least element of `l` is `None`. For instance
+```
+seq_options([Some(1), Some(2)])
+```
+will yield `Some([1, 2])`, but
+```
+seq_options([Some(1), Some(2), None])
+```
+will yield `None`.
+
+
+#### choose
+```
+Option.choose(o1 : option('a), o2 : option('a)) : option('a) 
+```
+
+Out of two `option`s choose the one that is `Some`, or `None` if both are `None`s.
+
+
+#### choose_first
+```
+Option.choose_first(l : list(option('a))) : option('a)
+```
+
+Same as [choose](#choose), but chooses from a list insted of two arguments.
+
+
+## String
```

Comment:
A lot of this isn't available until Iris

Reply:
marked.